projects/RabbitMQ.Client.OAuth2/OAuth2Client.cs

```diff
@@ -309,7 +309,7 @@ private Dictionary<string, string> BuildRefreshParameters(IToken token)
             Dictionary<string, string> dict = BuildRequestParameters();
             dict[GRANT_TYPE] = REFRESH_TOKEN;
 
-            if (_scope != null)
+            if (_scope != null && !dict.ContainsKey(SCOPE))
```

Comment:
Wouldn't in general the safer code be to replace the key?

Reply:
The whole check can probably be omitted entirely since the SCOPE dictionary key is already being set in BuildRequestParameters.